src/arrayMethodJoin.js

```diff
@@ -5,8 +5,22 @@
  *
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function() {
-    // write code here
+  [].__proto__.join2 = function(arg) {
+    return this.reduce((str, item) => {
```

Comment:
using reduce is overkill here. You can just iterate over array elements and concatenate the result.